compiler/modules/front/resolveTypes.bal

```diff
@@ -148,9 +148,16 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
             }
             t:ListDefinition d = new;
             td.defn = d;
-            t:SemType[] members = from var x in td.members select check resolveTypeDesc(mod, modDefn, depth + 1, x);
-            t:SemType rest = check resolveTypeDesc(mod, modDefn, depth + 1, td.rest);
-            return d.define(env, members, rest);
+            var tdLen = td.length?.value;
```

Comment:
Don't use `var` in a case like this: it makes the code harder to understand.

This bit is really ugly and repetitive.